src/main_namespace.h

```diff
@@ -0,0 +1,21 @@
+
+
+#ifndef MAINNAMESPACE
+#define MAINNAMESPACE
```

Comment:
This is very strange to need a main namespace, I suspect there are better ways of going about what is trying to be accomplished.